plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroTokenManager.java

```diff
@@ -0,0 +1,141 @@
+        final String usernamePassword = new String(rawEncoded, UTF_8);
+        final String header = "Basic " + usernamePassword;
+        BasicAuthToken token = new BasicAuthToken(header);
+        shiroTokenPasswordMap.put(token, password);
+
+        return token;
+    }
+
+    @Override
+    public boolean validateToken(AuthToken token) {
+        if (token instanceof BasicAuthToken) {
+            final BasicAuthToken basicAuthToken = (BasicAuthToken) token;
+            return basicAuthToken.getUser().equals(SecurityUtils.getSubject().toString())
+                && basicAuthToken.getPassword().equals(shiroTokenPasswordMap.get(basicAuthToken));
+        }
+        return false;
+    }
+
+    @Override
+    public String getTokenInfo(AuthToken token) {
+        if (token instanceof BasicAuthToken) {
+            final BasicAuthToken basicAuthToken = (BasicAuthToken) token;
+            return basicAuthToken.toString();
+        }
+        throw new UnsupportedAuthenticationToken();
+    }
+
+    @Override
+    public void revokeToken(AuthToken token) {
+        if (token instanceof BasicAuthToken) {
+            final BasicAuthToken basicAuthToken = (BasicAuthToken) token;
+            basicAuthToken.revoke();
+            return;
+        }
+        throw new UnsupportedAuthenticationToken();
+    }
+
+    @Override
+    public void resetToken(AuthToken token) {
+        if (token instanceof BasicAuthToken) {
+            final BasicAuthToken basicAuthToken = (BasicAuthToken) token;
+            basicAuthToken.revoke();
+        }
+    }
+
+    /**
+     * When the ShiroTokenManager is in use, a random password is generated for each token and is then output to the logs.
+     * The password is used for development only.
+     * @return A randomly generated password for development
+     */
+    public String generatePassword() {
+
+        CharacterRule lowercaseCharacterRule = new CharacterRule(EnglishCharacterData.LowerCase, 1);
+        CharacterRule uppercaseCharacterRule = new CharacterRule(EnglishCharacterData.UpperCase, 1);
+        CharacterRule numericCharacterRule = new CharacterRule(EnglishCharacterData.Digit, 1);
+        CharacterRule specialCharacterRule = new CharacterRule(EnglishCharacterData.Special, 1);
+
+        List<CharacterRule> rules = Arrays.asList(
+            lowercaseCharacterRule,
+            uppercaseCharacterRule,
+            numericCharacterRule,
+            specialCharacterRule
+        );
+        PasswordGenerator passwordGenerator = new PasswordGenerator();
+
+        Random random = Randomness.get();
+
+        String password = passwordGenerator.generatePassword(random.nextInt(8) + 8, rules); // Generate a 8 to 16 char password
+        log.info("Generated password: " + password);
+        return password;
+    }
+
+    public Map<BasicAuthToken, String> getShiroTokenPasswordMap() {
```

Comment:
Let's make it package private, seems to be for testing only, right?

```suggestion
Map<BasicAuthToken, String> getShiroTokenPasswordMap() {
```